pkg/artifactory/resource_artifactory_access_token.go

```diff
@@ -0,0 +1,111 @@
+package artifactory
+
+import (
+	"context"
+	"net/http"
+	"strconv"
+
+	"github.com/atlassian/go-artifactory/v2/artifactory"
+	v1 "github.com/atlassian/go-artifactory/v2/artifactory/v1"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceArtifactoryAcessToken() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAccessTokenCreate,
+		Read:   resourceAccessTokenRead,
+		Delete: resourceAccessTokenDelete,
+
+		Schema: map[string]*schema.Schema{
+			"username": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"scope": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"expires_in": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  3600,
+				ForceNew: true,
+			},
+			"refreshable": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+				ForceNew: true,
+			},
+			"audience": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"access_token": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"refresh_token": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func unmarshalToken(s *schema.ResourceData) (*v1.AccessTokenOptions, error) {
+	d := &ResourceData{s}
+
+	accessTokenOption := new(v1.AccessTokenOptions)
+
+	accessTokenOption.Username = d.getStringRef("username", false)
+	accessTokenOption.Scope = d.getStringRef("scope", false)
+	accessTokenOption.ExpiresIn = d.getIntRef("expires_in", false)
+	refreshable := d.getBoolRef("refreshable", false)
+	accessTokenOption.Refreshable = artifactory.String(strconv.FormatBool(*refreshable))
+	accessTokenOption.Audience = d.getStringRef("audience", false)
+
+	return accessTokenOption, nil
+}
+
+func resourceAccessTokenCreate(d *schema.ResourceData, m interface{}) error {
+	c := m.(*ArtClient).ArtOld
+
+	accessTokenOption, err := unmarshalToken(d)
```

Comment:
unhandled error

Reply:
@kuritka this method matches the Terraform contract defined for [`CreateFunc`](https://github.com/hashicorp/terraform/blob/ce9e4b2ebdd68609b2e9d34866bafb1c3c773109/helper/schema/resource.go#L199)

Reply:
@donovanmuller sorry for the previous comment, I was thinking `unmarshalToken` function which returns always nil, so we can remove returned error. 

line 77 would look like 
```
accessTokenOption := unmarshalToken(d) 
```
The function contract you mentioned is related to `resourceAccessTokenCreate`

Reply:
@kuritka got you, have adjusted.